### PR TITLE
Migrate dataset extra files store to Pinia

### PR DESCRIPTION
**Pull request by davelopez**
Requires #17119

Replaces the old Vuex store with a Pinia store and converts related files to Typescript.

## How to test the changes?
- [x] This is a refactoring of components with existing test coverage.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/datasets.py [hunk 1]
```diff
@@ -241,17 +242,30 @@ def update_permissions(
 
     @router.get(
         "/api/histories/{history_id}/contents/{history_content_id}/extra_files",
-        summary="Generate list of extra files.",
+        summary="Get the list of extra files/directories associated with a dataset.",
         tags=["histories"],
+        deprecated=True,
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/datasets.py]: I wouldn't deprecate this but I suppose it is an unpopular opinion.

**Review by jmchilton — APPROVED** [review]
Looks good!

File: lib/galaxy/webapps/galaxy/api/datasets.py (hunk 1 above)
Comment (davelopez) [reply lib/galaxy/webapps/galaxy/api/datasets.py]: I thought it was odd, that you need to put a history ID that is not used and will get ignored anyway :sweat_smile:
Also, as far as I know, this applies only to datasets and not to collections.
But I will remove the deprecation flag, I should have included it in the PR description, my bad :sweat_smile:

**Comment by davelopez** [comment]
Thank you @jmchilton! I removed the deprecation flag, you are right there is no need to deprecate it.

**Pull request merged**